schoolmatch.py

```diff
@@ -13,152 +20,171 @@
 import pandas as pd
 import numpy as np
 
-OSPI_df = pd.read_csv('Washington_School_Directory_20171121.csv')
-OSPI_Ind_df = pd.read_csv('ApprovedPrivateSchoolsList.csv')
 
-RS_df_pre = pd.read_csv('RSSchoolList2.csv', usecols=[0,1,2,3,4])
-RS_df_NotPub = RS_df_pre[RS_df_pre.Type != "Public"]
-RS_df = RS_df_pre[RS_df_pre.Type == "Public"]
+def clean_title(text):
+    """
+    Takes a pandas series of strings, and returns a pandas series of strings
+    that have been stripped and each word capitalized
+    """
+    text = text.str.lower()
+    text = text.str.title()
+    text = text.str.strip()
+    return text
+
+
+def string_truncate(strings_to_truncate, words_to_remove, truncated_list_name):
+    """
+    Takes a list of strings to modify, a list of words to remove from the first list,
+    and an empty list that will be populated with with the new truncated strings.
+    """
+    truncated_list_name = []
+    for string in strings_to_truncate:
+        modifying_string = string
+        for word in words_to_remove:
+            modifying_string = modifying_string.replace(word, "")
+            modifying_string = modifying_string.strip()
+        truncated_list_name.append(modifying_string)  
+    return truncated_list_name
+#Will take a look at stop words next
+
+
+#read in all csvs and create relevant dataframes
+ospi_raw_pub_df = pd.read_csv('Washington_School_Directory_20171121.csv')
+ospi_raw_ind_df = pd.read_csv('ApprovedPrivateSchoolsList.csv')
+ospi_raw_ind_df.dropna(how='all', axis=1, inplace=True)
+ospi_raw_ind_df.dropna(how='all', axis=0, inplace=True)
+
+
+rs_raw_df = pd.read_csv('RSSchoolList2.csv', usecols=[0,1,2,3,4])
+rs_ind_df = rs_raw_df[rs_raw_df.Type != "Public"].copy()
+rs_pub_df = rs_raw_df[rs_raw_df.Type == "Public"].copy()
+
+
+
+#clean all of the city names as they are obviously variable among sources
+ospi_raw_pub_df.City = clean_title(ospi_raw_pub_df.City) 
+ospi_raw_ind_df.City = clean_title(ospi_raw_ind_df.City)
 
+rs_pub_df.City = clean_title(rs_pub_df.City)
+rs_ind_df.City = clean_title(rs_ind_df.City)
 
 
-RSSchoolList = RS_df.Name.tolist()
-RSSchoolListStrip = []
-RSSchoolListBones = []
 
-RSIndSchoolList = RS_df_NotPub.Name.tolist()
-RSIndSchoolListStrip = []
-RSIndSchoolListBones = []
+#Initialize empty variables to be used later

```

Comment:
For lines 74-87.  Don't do this all at once.  Do this when needed. 
Most of these aren't needed.   In Python, you don't have to initialize variables, in general, unless you are going to try to append to it for example in a for loop.

Reply:
Yeah this confused me a bit as I thought I could define them all as I went, but I was getting errors saying the lists weren't defined, and this was my solution. I think a good example of what I saw was with string_truncate. I believe now that I was getting the error because the called function was trying to use an undefined variable as an argument. After removing that as a formal parameter, this works.